### PR TITLE
SOUNDS: Use 16 bits per sample

### LINKED ISSUE #1514: SOUNDS: 8 bits per sample sounds bad
```
const test_sound = sine_sound(400, 0.4);

const soft_sound = make_sound(t => get_wave(test_sound)(t) / 2 / 2 / 2 / 2 / 2 / 2, get_duration(test_sound));
play(soft_sound);
```

Something like this sounds very bad with 8 bits per sample, the (default) setting we're currently using for riffwave.js

There's a reason why most DAWs don't even support 8 bits per sample -- it sounds very bad when sounds get soft.

### PR DESCRIPTION
**Pull request by thomastanck**
### Description

Makes sounds sound less bad.

Fixes #1514 

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Code quality improvements

### How to Test

Deploy locally and test following code on playground with sounds library enabled:

```
const test_sound = sine_sound(400, 0.4);

const soft_sound = make_sound(t => get_wave(test_sound)(t) / 2 / 2 / 2 / 2 / 2 / 2, get_duration(test_sound));
play(soft_sound);
```

It should not sound terrible (like a sawtooth/square wave) and instead should just sound like a soft sine wave.

### Checklist

Please delete options that are not relevant.

- [x] I have tested this code

### CODE REVIEW

File: public/externalLibs/sound/sounds.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ function discretize_from(wave, duration, elapsed_duration, sample_length, data)
 // Quantize real amplitude values into standard 4-bit PCM levels
 function quantize(data) {
     for (var i = 0; i < data.length; i++) {
-        data[i] = Math.round((data[i] + 1) * 126);
+        data[i] = Math.round(data[i] * 32767);
```
Comment (martin-henz) [inline public/externalLibs/sound/sounds.js:63]: why not "+ 1"? what is the range for the result?

File: public/externalLibs/sound/sounds.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ function discretize_from(wave, duration, elapsed_duration, sample_length, data)
 // Quantize real amplitude values into standard 4-bit PCM levels
```
Comment (martin-henz) [inline public/externalLibs/sound/sounds.js:60]: should say 16-bit?

**Comment by thomastanck** [comment]
Fixed.

File: public/externalLibs/sound/sounds.js (hunk 1 above)
Comment (thomastanck) [reply public/externalLibs/sound/sounds.js]: The range is [-32768, 32767] (see riffwave.js)

(added a comment as well)

**Review by martin-henz — APPROVED** [review]

**Pull request merged**